# Epigenetics

This file contains the details on the epigenetics analysis in the project. The raw data consisted in excel tables, and the data was analysed using [Champ](https://github.com/YuanTian1991/ChAMP), an R package that was installed using:

    git clone https://github.com/YuanTian1991/ChAMP.git
    R CMD INSTALL ChAMP
    
More information on the tool ChAMP can be found in this [publication](https://academic.oup.com/bioinformatics/article/33/24/3982/4082274). The analyses run using this tool can be found in the R script "epigenetics.R" in this repository. The script's steps are:
1. **Load** data and **filter** based on p-value, missing data and so on. A number of 422520 probes were removed, leaving 422520 probes and 22 samples
2. **Plot** the data to check the quality. The plots were created per trait, so there is one for age, other for BMI, other for smoker status and so on.
3. **Data imputation**: all NA are imputed by KNN method with parameter k as 5
4. **Plot** the data to check the quality, now outputting a heatmap that allows to see grouping between samples, that later on can be used to draw conclusions about the population
5. **Data normalization**: the data is normalised according to the BMIQ method, which is the recomended for data that follows the beta distribution (like our data). This transformation is necesary because the Illumina array uses two types of hybridations (type I and type II), and the technical differences affect the data distribution, causing artifacts in the comparison of the two types of data. 
6. **Differential Methylation Probes**: to calculate differential methylation and check the results in a set of plots. It supports numeric variables like BMI and age, and categorical variables which contain more than two phenotypes (non-smoker, smoker and ex-smoker). "*The output includes some data frames of P-value, t-statistic and difference in mean methylation (this is labeled as logFC since it is analogous to the log fold-change in gene expression analysis) between two groups (for categorical covariate only). It also includes the annotation for each probe, the average beta value for the sample group, and the delta beta value for the two groups used in the comparison (delta value is same as logFC, we kept is here because very old version ChAMP used it).*" (extracted from ChAMP's manual)  

The R script used for the analysis can be found in this repository under the name ```5.1-EpigeneticAnalysis.R``` 
